src/ProductConstructionService/ProductConstructionService.DependencyFlow/SubscriptionTriggerer.cs

```diff
@@ -145,18 +149,25 @@ await AddDependencyFlowEventAsync(
             })
             .ToList();
 
-        _logger.LogInformation("Running asset update for {subscriptionId}", _subscriptionId);
-
-        await pullRequestActor.UpdateAssetsAsync(
-            _subscriptionId,
-            subscription.SourceEnabled
-                ? SubscriptionType.DependenciesAndSources
-                : SubscriptionType.Dependencies,
-            build.Id,
-            build.GitHubRepository ?? build.AzureDevOpsRepository,
-            build.Commit,
-            assets);
-
-        _logger.LogInformation("Asset update complete for {subscriptionId}", _subscriptionId);
+        await _redisMutex.ExecuteWhenReady(
```

Comment:
Usually, the conventions for these methods is `Enter`, `Wait`... I think we should follow that because it's unclear what "ready" means. It's not really ready - it's that no one else is holding the lock

Reply:
improved